### ISSUE TITLE
Null Pointer Exception when consuming a service token

### ISSUE THREAD

**Issue by zfi** [parallaxinc/Cloud-session-java-client#8]
An unhandled Null Pointer Exception occurs when attempting to validate the available token count for compile activities.

```
Jun 27, 2017 2:14:29 PM org.apache.catalina.core.StandardWrapperValve invoke
SEVERE: Servlet.service() for servlet [default] in context with path [/blockly] threw exception [java.lang.NullPointerException] with root cause
java.lang.NullPointerException
        at com.parallax.client.cloudsession.CloudSessionBucketService.handleResponse(CloudSessionBucketService.java:111)
        at com.parallax.client.cloudsession.CloudSessionBucketService.consumeOne(CloudSessionBucketService.java:53)
        at com.parallax.server.blocklyprop.rest.RestCompile.checkLimiterBucket(RestCompile.java:144)
        at com.parallax.server.blocklyprop.rest.RestCompile.compileC(RestCompile.java:106)
        at sun.reflect.GeneratedMethodAccessor48.invoke(Unknown Source)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        [...]
```

**Comment by zfi** [comment]
The CloudSessionBucketService.consumeOne() attempts to decrement the number of available compile tokens by one. If there are no tokens available, The Cloud Session service should return a payload containing error code 470, a message and a time stamp that indicates when the next token will become available. 

The Cloud Session service was omitting the time stamp from the payload. There was no check in the CloudSessionBucketService.handleResponse() for a missing timestamp. The attempt to convert the non-existing timestamp string to a datetime value was causing the Null Pointer exception. 

Adding an issue to the Cloud Session service project to correct this.

**Comment by PropGit** [comment]
Great find @zfi!

**Comment by zfi** [comment]
Fixed in Cloud Session Server with commit 9e59ee88b70faa256d261c29501a8f7f50715f4e.
